### PR TITLE
Makefile: add deprecation warning and alternate suggestions

### LINKED ISSUE #75734: Makefile: add warnings advising people to use `dev` in place of `make build`, `make test`, etc.
We don't have to cover every single use case for the `Makefile`, but covering the following (and smartly recommending the corresponding `dev` command) would be a good start:

```
make build{,short}
make test
make lint{,short}
... others?
```

Epic CRDB-8036

### LINKED ISSUE #84362: dev: symlink `githooks/` through `dev doctor`
This is something the Makefile currently does, which primes the commit message template among other things:

https://github.com/cockroachdb/cockroach/blob/ae220abff4e90a056c3f0c7b173eab01aedb2960/Makefile#L350-L356

Jira issue: CRDB-17617

### PR DESCRIPTION
**Pull request by irfansharif**
**dev: symlink githooks/ into .git/hooks**

Fixes https://github.com/cockroachdb/cockroach/issues/84362, doing something the Makefile previously did.

---

**Makefile: add deprecation warning and alternate suggestions**

Closes #75734. We now shout the following at the very top:
```
  [WARNING] Makefile is deprecated and will be removed shortly.
```
For common subcommands, we provide best-effort hints for the dev alternatives:
```
  $ make build
  [WARNING] Use `dev build cockroach` instead.

  $ make buildshort
  [WARNING] Use `dev build cockroach-short` instead.

  $ make test PKG=./pkg/sql TESTS='TestCascadingZoneConfig' TESTFLAGS=-v
  [WARNING] Use `dev test pkg/sql --filter TestCascadingZoneConfig --test-args "-v"` instead.

  $ make stressrace PKG=./pkg/spanconfig/spanconfigmanager TESTS=TestManagerCheckJobConditions TESTFLAGS='-v -show-logs'
  [WARNING] Use `dev test --stress --race pkg/spanconfig/spanconfigmanager --filter TestManagerCheckJobConditions --test-args "-v -show-logs" --timeout 45m` instead.

  $ make testccllogic FILES='zone_config'
  [WARNING] Use `dev test pkg/ccl/logictestccl  --filter Test(CCL|Tenant)Logic//^zone_config$/ --test-args "-test.v -show-sql "` instead.

  $ make bench PKG=./pkg/spanconfig/spanconfigkvsubscriber BENCHES='BenchmarkSpanConfigDecoder' TESTFLAGS='-v -count 2 -benchtime=10x'
  [WARNING] Use `dev bench pkg/spanconfig/spanconfigkvsubscriber --filter BenchmarkSpanConfigDecoder --timeout 5m --test-args "-v -count 2 -benchtime=10x"` instead.

  $ make lint TESTS=TestLint/CODEOWNERS
  [WARNING] Use `dev lint  --filter TestLint/CODEOWNERS --timeout 30m` instead.

  $ make lintshort TESTS=TestGolint
  [WARNING] Use `dev lint --filter TestGolint --timeout 30m` instead.

  $ make acceptance TESTS=TestDockerC PKG=./pkg/acceptance TESTFLAGS=-v
  [WARNING] Use `dev acceptance pkg/acceptance --filter TestDockerC --timeout 30m --test-args "-v"` instead.

  $ make generate
  [WARNING] Use `dev generate protobuf` instead.
  [WARNING] Use `dev generate` instead.
  [WARNING] Use `dev generate execgen` instead.

  $ make protobuf
  [WARNING] Use `dev generate protobuf` instead.

  $ make bazel-generate
  [WARNING] Use `dev generate bazel` instead.

  $ make ui-maintainer-clean
  [WARNING] Use `dev ui clean` instead.
  [WARNING] Use `dev ui clean --all` instead.
```
Release note: None

### CODE REVIEW

File: pkg/cmd/dev/doctor.go [hunk 1]
```diff
@@ -195,13 +196,31 @@ You can install node with: `+"`pkg install node`")
 		}
 	}
 
+	d.log.Println("doctor: running githooks check")
```
Comment (rickystewart) [inline pkg/cmd/dev/doctor.go:199]: Is this stuff supposed to be in a different PR? If not, mentioning it in the PR description would be good. :P

**Review by rickystewart — APPROVED** [review]
Awesome!

File: pkg/cmd/dev/doctor.go (hunk 1 above)
Comment (irfansharif) [reply pkg/cmd/dev/doctor.go:199]: Done.
